resources/assets/lib/beatmapsets-show/extra.tsx

```diff
@@ -0,0 +1,139 @@
+    };
+  }
+
+  componentDidMount() {
+    this.setState({ chartAreaBound: this.chartAreaRef.current?.getBoundingClientRect() ?? null });
+  }
+
+  render() {
+    const successRate = round((this.props.beatmap.passcount / this.props.beatmap.playcount) * 100, 1);
+    const ratings = this.getRatings();
+
+    return (
+      <div className='beatmapset-extra'>
+        <div className='beatmapset-extra__item'>
+          {osu.trans('beatmapsets.show.info.success-rate')}
+        </div>
+        <div className='beatmapset-extra__item'>
+          <div className='beatmapset-extra__right'>
+            {`${successRate}%`}
+          </div>
+        </div>
+        <div className='beatmapset-extra__item beatmapset-extra__item--bar'>
+          {this.renderBar(successRate)}
+        </div>
+
+        <div className='beatmapset-extra__item'>
+          {osu.trans('beatmapsets.show.stats.user-rating')}
+        </div>
+        <div className='beatmapset-extra__item beatmapset-extra__item--ratings'>
+          <div>
+            {ratings.negative}
+          </div>
+          <div className='beatmapset-extra__right'>
+            {ratings.positive}
+          </div>
+        </div>
+        <div className='beatmapset-extra__item beatmapset-extra__item--bar'>
+          {this.renderBar((ratings.positive * 100) / (ratings.positive + ratings.negative), true)}
+        </div>
+
+        {this.props.beatmapset.is_scoreable && (
+          <>
+            <div className='beatmapset-extra__item'>
+              {osu.trans('beatmapsets.show.stats.rating-spread')}
+            </div>
+            <div ref={this.chartAreaRef} className='beatmapset-extra__item beatmapset-extra__item--chart'>
+              {this.renderChart(this.props.beatmapset.ratings.slice(1))}
+            </div>
+          </>
+        )}
+      </div>
+    );
+  }
+
+  private getRatings = () => this.props.beatmapset.ratings.reduce(
+    (result, count, rating) => {
+      if (rating >= 1 && rating <= 5) {
+        result.negative += count;
+      } else if (rating >= 6 && rating <= 10) {
+        result.positive += count;
+      }
+      return result;
+    },
+    { negative: 0, positive: 0 },
+  );
+
+  private renderBar(fill: number, fromRight = false) {
+    return (
+      <div className='beatmapset-bar beatmapset-bar--beatmapset-extra'>
+        <div
+          className={classWithModifiers('beatmapset-bar__fill', {
+            right: fromRight,
+          })}
+          style={{ width: `${fill}%` }}
+        />
+      </div>
+    );
+  }
+
+  private renderChart(ratings: number[]) {
+    const areaWidth = this.state.chartAreaBound?.width ?? 0;
+    const areaHeight = this.state.chartAreaBound?.height ?? 0;
+
+    const spacing = 2;
+    const barCounts = ratings.length;
+    const barWidth = (areaWidth - ((barCounts - 1) * spacing)) / barCounts;
+
+    const maxRating = Math.max(...ratings);
+
+    return (
+      <svg height={areaHeight} width={areaWidth}>
```

Comment:
I tried using div before this but couldn't find how to make background gradient across multiple divs like in figma.

![Screenshot from 2021-08-14 13-59-07](https://user-images.githubusercontent.com/4964985/129438080-f3e867c1-b9a6-42d5-99ff-2517674879db.png)

Reply:
[This is one way](https://codepen.io/nanaya/pen/vYmwaEO) (with size and position as inline style variable instead of modifier).

The other way is just to set solid background for each of them (with d3 gradient function). Should be convincing enough. (though in this case the colors will need to be re-defined in js)

Linear gradient using the same function for each bar also works.

Reply:
Thanks!